refapp/SKWebApi/Skills/ChatSkill.cs

```diff
@@ -0,0 +1,285 @@
+
+    [SKFunction("Extract user memories")]
+    [SKFunctionName("ExtractUserMemories")]
+    [SKFunctionContextParameter(Name = "tokenLimit", Description = "Maximum number of tokens")]
+    [SKFunctionContextParameter(Name = "contextTokenLimit", Description = "Maximum number of context tokens")]
+    public async Task<string> ExtractUserMemoriesAsync(SKContext context)
+    {
+        var tokenLimit = int.Parse(context["tokenLimit"], new NumberFormatInfo());
+        var contextTokenLimit = int.Parse(context["contextTokenLimit"], new NumberFormatInfo());
+
+        var remainingToken = Math.Min(
+            tokenLimit,
+            Math.Floor(contextTokenLimit * SystemPromptDefaults.MemoriesResponseContextWeight)
+        );
+
+        var latestMessage = await this.GetLatestMemoryAsync(context);
+        Console.WriteLine($"Latest message: {latestMessage.Text}");
+        var results = context.Memory.SearchAsync("ChatMessages", latestMessage.Text, limit: 1000);
+
+        string memoryText = "";
+        await foreach (var memory in results)
+        {
+            var estimatedTokenCount = this.EstimateTokenCount(memory.Text);
+            if (remainingToken - estimatedTokenCount > 0)
+            {
+                memoryText += $"\n{memory.Text}";
+                remainingToken -= estimatedTokenCount;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        return $"Past memories:\n{memoryText.Trim()}";
+    }
+
+    [SKFunction("Extract chat history")]
+    [SKFunctionName("ExtractChatHistory")]
+    [SKFunctionContextParameter(Name = "tokenLimit", Description = "Maximum number of tokens")]
+    [SKFunctionContextParameter(Name = "contextTokenLimit", Description = "Maximum number of context tokens")]
+    public async Task<string> ExtractChatHistoryAsync(SKContext context)
+    {
+        var tokenLimit = int.Parse(context["tokenLimit"], new NumberFormatInfo());
+
+        // TODO: Use contextTokenLimit to determine how much of the chat history to return
+        // TODO: relevant history
+
+        var remainingToken = tokenLimit;
+        string historyText = "";
+
+        await foreach (var message in this.GetAllMemoriesAsync(context))
+        {
+            var estimatedTokenCount = this.EstimateTokenCount(message.Text);
+            if (remainingToken - estimatedTokenCount > 0)
+            {
+                historyText += $"\n{message.Text}";
+                remainingToken -= estimatedTokenCount;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        return $"Chat history:\n{historyText.Trim()}";
+    }
+
+    [SKFunction("Get chat response")]
+    [SKFunctionName("Chat")]
+    [SKFunctionInput(Description = "The new message")]
+    [SKFunctionContextParameter(Name = "audience", Description = "The audience the chat bot is interacting with.")]
+    public async Task<SKContext> ChatAsync(string message, SKContext context)
```

Comment:
Is this the "top-level" SK function? If so, please add an XMLDoc comment stating that and explaining a bit on why.